src/bridging/types.ts

```diff
@@ -0,0 +1,174 @@
+import { latest as latestAppData } from '@cowprotocol/app-data'
+import { ethers } from 'ethers'
+import { ChainInfo, ChainId } from '../common'
+import { TokenInfo } from '../common/tokens'
+import { Address } from '../order-book'
+import { EvmCall } from '../common/transaction'
+
+export interface BridgeProviderInfo {
+  name: string
+  logoUrl: string
+}
+
+export interface WithSellToken {
+  sellTokenChainId: ChainId
+  sellTokenAddress: Address
+  sellTokenDecimals: number
+}
+
+export interface WithBuyToken {
+  buyTokenChainId: ChainId
+  buyTokenAddress: Address
+  buyTokenDecimals: number
+}
+
+export interface GetBuyTokensParams extends Partial<WithSellToken> {
+  targetChainId: ChainId
+}
+
+/**
+ * Parameters for getting a bridge quote
+ */
+export interface QuoteBridgeRequest extends WithSellToken, WithBuyToken {
+  type: 'SELL' // We make it explicit that only SELL is supported for now
+  amount: string
+  recipient: string
+}
+
+export interface QuoteFee {
+  type: 'fixed' | 'percent'
+  value: number
+}
+
+export interface BridgeQuoteResult {
+  fee: QuoteFee
+  slippageBps: number
+  buyAmount: string
+  fillTimeInSeconds?: number
+}
+
+export interface BridgeHook {
+  postHook: latestAppData.CoWHook
+  recipient: string
+}
+
+export enum BridgeStatus {
+  INITIATED = 'initiated',
+  EXECUTED = 'executed',
+  FAILED = 'failed',
+  EXPIRED = 'expired',
+}
+
+export interface BridgeStatusResult {
+  status: BridgeStatus
+  fillTimeInSeconds?: number
+}
+
+/**
+ * A bridge deposit. It includes the sell amount and the minimum buy amount.
+ *
+ * It models the minimum information needed to initiate a bridge and that it can also be extracted from the cow hook.
+ *
+ */
+export interface BridgeDeposit extends QuoteBridgeRequest {
+  provider: BridgeProviderInfo
+  minBuyAmount: string
+}
+
+export interface BridgeProvider<Q extends BridgeQuoteResult> {
+  info: BridgeProviderInfo
+
+  /**
+   * Get basic supported chains
+   */
+  getNetworks(): Promise<ChainInfo[]>
+
+  /**
+   * Get supported tokens for a chain
+   */
+  getBuyTokens(chainId: GetBuyTokensParams): Promise<TokenInfo[]>
+
+  /**
+   * Get intermediate tokens given a quote request.
+   *
+   * An intermediate token, is a token in the source chain, that could be used to bridge the tokens to the destination chain.
+   * This method returns a sorted list of tokens, they are sorted by priority, so first tokens are more likely to be more liquid.
+   *
+   * @param request - The quote request
+   */
+  getIntermediateTokens(request: QuoteBridgeRequest): Promise<string[]>
```

Comment:
Why the response is a string[] instead of TokenInfo[]?

Reply:
We don't need so much detail for the intermediate token. This method is not meant to be used by the UI, only by the SDK implementation. 

Its possible that as you say, we return a token info here, but until we do the real implementation we start from less data to more. In the prototype, with the string I have enough. I would only expose it if this needs to be exposed to the UI (possible we want to do that in the quote response), but the quote is not handled yet in this PR